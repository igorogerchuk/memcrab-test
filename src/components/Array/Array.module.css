.tableWrapper {
  margin: 0 auto;
  overflow: auto;
  height: 100vh;
}

.table {
  margin: 0 auto;
}

.addCell {
  background: linear-gradient(110deg, #eee 0%, #e3e3e3 10%, #fff 20%, #fff 25%, #f0f0f0 26%, #fff 28%, #ddd 55%, #eee 100%)
}

.addCell:active {
  position: relative;
  top: 2px;
}

td {
  border-radius: 15px;
  width: 45px;
  height: 40px;
  border: 1px solid #999;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.5), 0px 0px 5px rgba(0, 0, 0, 0.25), 0px 0px 2px 4px rgba(0, 0, 0, 0.1);
  color: #444;
  font-size: 16px;
  font-weight: bold;
  text-shadow: 0px 1px 1px rgba(0, 0, 0, 0.3)
}