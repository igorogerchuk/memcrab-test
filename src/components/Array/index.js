// @flow
export { default } from "./Array";
