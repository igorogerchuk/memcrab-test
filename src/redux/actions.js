// @flow

import * as types from "./types";
import type { ArrayState, RowsState, CellsState, ParamsState } from "./types";

type _ExtractReturn<R, F: (...args: any[]) => R> = R;
type ExtractReturn<F> = _ExtractReturn<*, F>;

export const removeRow = (id: string, cellsIds: $ReadOnlyArray<string>) => ({
  type: types.REMOVE_ROW,
  payload: { id, cellsIds }
});

export const addRow = (
  array: ArrayState,
  row: RowsState,
  cells: CellsState
) => ({
  type: types.ADD_ROW,
  payload: { array, row, cells }
});

export const saveArray = (array: ArrayState) => ({
  type: types.SAVE_ARRAY,
  payload: { array }
});

export const saveRows = (rows: RowsState) => ({
  type: types.SAVE_ROWS,
  payload: { rows }
});

export const saveCells = (cells: CellsState) => ({
  type: types.SAVE_CELLS,
  payload: { cells }
});

export const increase = (id: string) => ({
  type: types.INCREASE,
  payload: { id }
});

export const saveParams = (params: ParamsState) => ({
  type: types.SAVE_PARAMS,
  payload: { params }
});

export type Action =
  | ExtractReturn<typeof removeRow>
  | ExtractReturn<typeof addRow>
  | ExtractReturn<typeof saveArray>
  | ExtractReturn<typeof saveRows>
  | ExtractReturn<typeof saveCells>
  | ExtractReturn<typeof saveParams>
  | ExtractReturn<typeof increase>;
