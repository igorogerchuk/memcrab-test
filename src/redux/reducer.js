import types from "./types";

const reducer = (state = { array: [] }, { type, payload }) => {
  const { array } = state;
  switch (type) {
    case types.ADD_LINE:
      return {
        array: array.concat(payload.newLine)
      };
    case types.REMOVE_LINE:
      return {
        array: array.filter(line => line !== payload.line)
      };
    case types.SAVE_ARRAY:
      return { array: payload.array };
    default:
      return state;
  }
};

export default reducer;
