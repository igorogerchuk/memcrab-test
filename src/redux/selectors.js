// @flow
import { createSelector } from "reselect";
import type { State } from "./types";

export const getArray = (state: State) => state.array;

export const getRows = (state: State) => state.rows;

export const getCells = (state: State) => state.cells;

export const getColumnsQty = (state: State) => state.params.n;

export const getIlluminatedQty = (state: State) => state.params.x;

export const getSumColumn = createSelector<State, *, *, *, *>(
  [getRows, getCells],
  (rows, cells) => {
    const sumColumn = [];
    const cellsIds = Object.keys(rows).map((rowId: string) => rows[rowId]);
    for (let i = 0; i < cellsIds.length; i++) {
      let rowTotal = 0;
      for (let j = 0; j < cellsIds[i].length; j++) {
        rowTotal += cells[cellsIds[i][j]].amount;
      }
      sumColumn.push(rowTotal);
    }
    return sumColumn;
  }
);

export const getAvarageRow = createSelector<State, *, *, *, *>(
  [getRows, getCells],
  (rows, cells) => {
    const avarageRow = [];
    const cellsIds = Object.keys(rows).map((rowId: string) => rows[rowId]);
    for (let j = 0; j < cellsIds[0].length; j++) {
      let columnTotal = 0;
      for (let i = 0; i < cellsIds.length; i++) {
        columnTotal += cells[cellsIds[i][j]].amount;
      }
      avarageRow.push(columnTotal / cellsIds.length);
    }
    return avarageRow;
  }
);
