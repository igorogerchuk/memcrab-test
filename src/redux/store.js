// @flow
import { createStore } from "redux";
import reducer from "./reducer";
import type { State } from "./types";
import type { Dispatch } from "redux";
import type { Action } from "./actions";

const store = createStore<State, Action, Dispatch<Action>>(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
export default store;
