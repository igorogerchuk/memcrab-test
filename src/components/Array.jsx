import React, { Component } from "react";
import { connect } from "react-redux";
import * as selectors from "../redux/selectors";
import ArrayLine from "./ArrayLine";
import AvarageRow from "./AvarageRow";
const uuidv4 = require("uuid/v4");

class Array extends Component {
  render() {
    const { array } = this.props;
    return (
      <div>
        <table>
          <tbody>
            {array.map(line => {
              let id = uuidv4();
              return <ArrayLine key={id} id={id} line={line} />;
            })}
            <AvarageRow />
          </tbody>
        </table>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return { array: selectors.getArray(state) };
};

export default connect(mapStateToProps)(Array);
