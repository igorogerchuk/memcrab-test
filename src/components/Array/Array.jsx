import React, { Component } from "react";
import { connect } from "react-redux";
import ArrayLine from "../ArrayLine";
import AvarageRow from "../AvarageRow";
import styles from "./Array.module.css";
import * as selectors from "../../redux/selectors";
// import getClosestNumbers from "../../services/closestNumbers";

class Array extends Component {
  // state = { hover: false, closestNumbers: [] };

  // illuminateHandler = e => {
  // const { id } = e.target;
  // if (id.slice(0, 6) !== "number") {
  //   return;
  // }
  // const { array, numberQty } = this.props;
  // this.setState(state => {
  //   return {
  //     hover: !state.hover,
  //     closestNumbers: !state.hover
  //       ? getClosestNumbers(array, id, numberQty)
  //       : []
  //   };
  // });
  // };

  render() {
    const { array } = this.props;
    // const { closestNumbers } = this.state;

    return (
      <div>
        <table
          className={styles.matrix}
          // onMouseOver={this.illuminateHandler}
          // onMouseOut={this.illuminateHandler}
        >
          <tbody>
            {array.map(line => (
              <ArrayLine
                key={line.id}
                id={line.id}
                line={line.cells}
                // closestNumbers={closestNumbers}
              />
            ))}
            <AvarageRow />
          </tbody>
        </table>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    array: selectors.getArray(state)
    // numberQty: selectors.getNumbersQty(state)
  };
};

export default connect(mapStateToProps)(Array);
