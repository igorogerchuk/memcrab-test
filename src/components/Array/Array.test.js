import React from "react";
import Array from "./Array";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../../redux/reducer";
import createRandomArray from "../../services/randomArray";
import uuid from "uuid/v4";

it("should render correctly with initial state from Redux store", () => {
  const store = createStore(reducer);
  const snap = renderer
    .create(
      <Provider store={store}>
        <Array />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});

jest.mock("uuid/v4");
let i = 1;
uuid.mockImplementation(() => i++);
jest.spyOn(Math, "random").mockReturnValue(0.5);
const { array, rows, cells } = createRandomArray(5, 4);
const params = { n: 4, x: 3 };
const store = createStore(reducer, { array, rows, cells, params });

it("should render correctly with given random array 5x4", () => {
  const snap = renderer
    .create(
      <Provider store={store}>
        <Array />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});

it("should render correctly after click on the addButton", () => {
  let component;
  renderer.act(() => {
    component = renderer.create(
      <Provider store={store}>
        <Array />
      </Provider>
    );
  });

  const instance = component.root;
  const addButton = instance.findByProps({
    id: "addButton",
  });
  renderer.act(() => addButton.props.onClick());
  const snap = component.toJSON();
  expect(snap).toMatchSnapshot();
});
