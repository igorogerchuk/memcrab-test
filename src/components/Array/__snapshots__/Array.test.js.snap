// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly after click on the addButton 1`] = `
<div
  styleName="tableWrapper"
>
  <table
    styleName="table"
  >
    <tbody>
      <tr
        id={1}
      >
        <td
          id={2}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={4}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={5}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={1}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={6}
      >
        <td
          id={7}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={8}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={9}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={10}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={6}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={11}
      >
        <td
          id={12}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={13}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={14}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={15}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={11}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={16}
      >
        <td
          id={17}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={18}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={19}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={20}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={16}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={21}
      >
        <td
          id={22}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={23}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={24}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={25}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={21}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={26}
      >
        <td
          id={27}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={28}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={29}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={30}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={26}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
      </tr>
      <tr>
        <td
          colSpan={4}
          styleName="addCell"
        >
          <button
            id="addButton"
            onClick={[Function]}
            styleName="addButton"
          >
            +
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render correctly with given random array 5x4 1`] = `
<div
  styleName="tableWrapper"
>
  <table
    styleName="table"
  >
    <tbody>
      <tr
        id={1}
      >
        <td
          id={2}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={3}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={4}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={5}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={1}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={6}
      >
        <td
          id={7}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={8}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={9}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={10}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={6}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={11}
      >
        <td
          id={12}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={13}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={14}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={15}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={11}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={16}
      >
        <td
          id={17}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={18}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={19}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={20}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={16}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr
        id={21}
      >
        <td
          id={22}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={23}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={24}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          id={25}
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "background": "linear-gradient(110deg,#eee 0%,#e3e3e3 10%,#fff 20%,#fff 25%,#f0f0f0 26%,#fff 28%,#ddd 55%,#eee 100%)",
            }
          }
          styleName="simpleTd"
        >
          500
        </td>
        <td
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          styleName="sumCell"
        >
          2000
        </td>
        <td
          styleName="removeButtonTd"
        >
          <button
            id={21}
            onClick={[Function]}
            styleName="removeButton"
          >
            ×
          </button>
        </td>
      </tr>
      <tr>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
        <td
          styleName="avarageTd"
        >
          500.00
        </td>
      </tr>
      <tr>
        <td
          colSpan={4}
          styleName="addCell"
        >
          <button
            id="addButton"
            onClick={[Function]}
            styleName="addButton"
          >
            +
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render correctly with initial state from Redux store 1`] = `null`;
